controllers/discovery/initDiscoveryUserData.js

```diff
@@ -41,7 +54,7 @@ const InitDiscoveryUserData = async (req, res) => {
             A.karma_score,
             ARRAY( select name from topics as tp
               left join user_topics as utp on tp.topic_id = utp.topic_id
-              where utp.user_id = A.user_id limit 3
+              where utp.user_id = A.user_id and tp.topic_id in (:topicIds) limit 3
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

This line introduces a filter in the community info based on the user's topics which is good. However, you should consider handling SQL Injection attacks by ensuring that `topicIds` is properly sanitized before being used in the query.



<!-- This is an auto-generated comment by OSS CodeRabbit -->